share/shwap/p2p/bitswap/block_registry.go

```diff
@@ -0,0 +1,29 @@
+package bitswap
+
+import (
+	"hash"
+
+	"github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// RegisterBlock registers the new Block type and multihash for it.
+func RegisterBlock(mhcode, codec uint64, size int, bldrFn func(cid.Cid) (Block, error)) {
```

Comment:
Should this be exported?

Reply:
It depends on our strategy and whether we want shwap users to register their own block types without forking. One part of me likes the extensibility we provide here and wants to make it open for everyone, but the other side tells me it is pretty pointless and would require adding a new message across the stack vertically anyway, not only here. 

I am inclined toward the latter, and we can always so will privatize this. Good call!